### PR TITLE
[SDTEST-477] add command line tool to compute a percentage of skippable tests for RSpec

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
Adds a command line interface `ddcirb` that outputs percentage of skipped tests for the current test run. 

Documentation on the usage is provided within PR in [CommandLineInterface.md](https://github.com/DataDog/datadog-ci-rb/pull/194/files#diff-4d14425a7f9dbebe76a2294079b440fda2ef677c3d1ac672e74498066f653328) file.

**Motivation**
This tool is useful to determine a number of parallel jobs when running big rspec test suites.

**How to test the change?**
Tested using https://github.com/anmarchenko/forem project.

### CODE REVIEW

File: lib/tasks/rspec.rake [hunk 1]
```diff
@@ -0,0 +1,58 @@
+namespace :datadog do
+  namespace :ci do
+    namespace :rspec do
+      task :skippable_percentage do
```
Comment (devinburnette) [inline lib/tasks/rspec.rake:4]:
```suggestion
      task skippable_percentage: :environment do
```

File: lib/tasks/rspec.rake [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        require "rspec/core"
+        require "datadog/ci"
+
+        Datadog.configure do |c|
+          c.ci.enabled = true
+          c.ci.itr_enabled = true
+          c.ci.discard_traces = true
+          c.ci.instrument :rspec, dry_run_enabled: true
```
Comment (devinburnette) [inline lib/tasks/rspec.rake:12]:
```suggestion
          c.ci.instrument :rspec, dry_run_enabled: true
          c.tracing.enabled = true
```

File: lib/tasks/rspec.rake [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        end
+
+        rspec_cli_options = %w[
+          --dry-run
```
Comment (devinburnette) [inline lib/tasks/rspec.rake:16]:
```suggestion
          --dry-run
          --require
          rails_helper
```
(or spec_helper)

File: lib/tasks/rspec.rake [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        if exit_code != 0
+          Datadog.logger.error("RSpec dry-run failed with exit code #{exit_code}")
+        end
+
+        itr = Datadog.send(:components).itr
+        print((itr.skipped_tests_count.to_f / itr.total_tests_count).floor(2))
+      end
+
+      task :skippable_percentage_estimate do
```
Comment (devinburnette) [inline lib/tasks/rspec.rake:31]:
```suggestion
      task skippable_percentage_estimate: :environment do
```

File: lib/tasks/rspec.rake [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        if ENV["DD_SERVICE"].nil?
+          Datadog.logger.error("DD_SERVICE is not set. You must provide it to estimate the skippable percentage.")
+          exit 1
+        end
+
+        Datadog.configure do |c|
+          c.ci.enabled = true
+          c.ci.itr_enabled = true
+          c.ci.discard_traces = true
```
Comment (devinburnette) [inline lib/tasks/rspec.rake:42]:
```suggestion
          c.ci.discard_traces = true
          c.tracing.enabled = true
```

File: lib/datadog/ci/test_optimisation/component.rb [hunk 6]
```diff
@@ -156,16 +158,18 @@ def mark_if_skippable(test)
...
         def count_skipped_test(test)
-          return if !test.skipped? || !test.skipped_by_itr?
-
-          if forked?
-            Datadog.logger.warn { "Intelligent test runner is not supported for forking test runners yet" }
-            return
-          end
-
           @mutex.synchronize do
             Telemetry.itr_skipped
```
Comment (anmarchenko) [inline lib/datadog/ci/test_optimisation/component.rb:162]: FIXME: this is a bug, need to be moved down to count skipped tests correctly

File: docs/CommandLineInterface.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+- `bundle exec ddcirb skipped-tests` - outputs the percentage of skipped tests to stdout. Note that it runs your specs
+in dry run mode, don't forget to set RAILS_ENV=test environment variable.
+- `bundle exec ddcirb skipped-tests-estimate` - estimates the percentage of skipped tests and outputs to stdout without loading
+your test suite and running it in dry run mode. ATTENTION: this is considerably faster but could be very inaccurate.
+
+Example usage:
+
+```bash
+$ RAILS_ENV=test bundle exec ddcirb skipped-tests
```
Comment (juan-fernandez) [inline docs/CommandLineInterface.md:21]: should we show an example of the output? Or is that somewhere else?

**Review by juan-fernandez — APPROVED** [review]

File: docs/CommandLineInterface.md (hunk 7 above)
Comment (anmarchenko) [reply docs/CommandLineInterface.md:21]: the output just one line below

Comment (anmarchenko) [reply docs/CommandLineInterface.md:21]: 0.45 is the output

**Pull request merged**